### ISSUE TITLE
gaia-switch needs love

### ISSUE THREAD

**Issue by gmarty** [gaia-components/gaia-switch#20]
This component is AFAICT only used in gaia-theme. On the other hand, a lot of Gaia apps use the gaia-switch from Gaia tree (`shared/elements/gaia_switch`).
gaia_switch is used but untested and inside Gaia.

We should reconciliate and consolidate both components into a single one.

**Comment by wilsonpage** [comment]
This <gaia-switch> was used inside some of the Spark apps IIRC. What would you like to see done for this issue to be closed?

**Comment by gmarty** [comment]
Merge in the component from gaia_switch, adapt unit tests and check it in gaia instead of the previous one.
How does it sound?

**Comment by wilsonpage** [comment]
Ah, so you're saying gaia_switch has unit-tests that (this) <gaia-switch> doesn't have? I'm confused what goodness gaia_switch has over (this) <gaia-switch>.

**Comment by gmarty** [comment]
Nah, this one has unit tests, but the other one is more widely used, so we don't want to potentially regress.
Disclaimer: I haven't taken a look at the code so that's assuming both components are idempotent.
Apart from that I'll be happy with any solution fixing the duplication here.

**Comment by wilsonpage** [comment]
So this bug asking to replace all instances of gaia_switch with gaia-switch in Gaia?

**Comment by gmarty** [comment]
This bug is about removing the duplication of a similar component on 2 different sources, whatever the solution we eventually go for.

**Comment by wilsonpage** [comment]
So you're asking for changes to be made in Gaia repo?
